File: code-ninja-course/src/main/java/com/codeninja/solidsolution/universities/University.java
package com.codeninja.solidsolution.universities;

import com.codeninja.solidsolution.*;
import com.codeninja.solidsolution.exceptions.NotImplemented;
import com.codeninja.solidsolution.students.Student;

public class University {
	int ID;
	
	public Student createStudent(String emailAddress) {
		return new Student(emailAddress, ID);
	}
	
	public int getID() {
		return ID;
	}

	public PackageType getPackageType() {
		throw new NotImplemented();
	}
	private void setID(int ID) {
		throw new NotImplemented();
	}
	private void setName(String name) {
		throw new NotImplemented();
	}
	
	private void setPackageType(PackageType packageType) {
		throw new NotImplemented();
	}

}

File: code-ninja-course/CSHARP/Code/SOLID/University.cs
using System;

namespace LegacyCodeToTestableCode
{
    public class University
    {
        int _ID=0;

        public int GetID()
        {
            return _ID;
        }

        public string GetName()
        {
            throw new NotImplementedException();
        }

        public PackageType GetPackageType()
        {
            throw new NotImplementedException();
        }

        private void SetID(int id)
        {
            throw new NotImplementedException();
        }
        private void SetName(string name)
        {
            throw new NotImplementedException();
        }

        private void SetPackageType(PackageType packageType)
        {
            throw new NotImplementedException();
        }


    }
}